crates/dns-resolver/src/resolver.rs

```diff
@@ -47,79 +48,121 @@ impl Answer {
     }
 }
 
-pub enum Resolver {
-    Tokio(TokioAsyncResolver),
-    #[cfg(feature = "unbound")]
-    Unbound(AsyncContext),
+#[async_trait]
+pub trait Resolver: Send + Sync + 'static {
```

Comment:
I think that the shape of this trait looks right. I was thinking that this might end up being extracted into a `dns-resolver-trait` crate that is the minimal set of deps just for this, and that has some support for mocking/providing hard coded zone information as we discussed earlier.  Is that the ultimate plan?

Reply:
As mentioned in the OP, figured I might do an `impl` feature for the dns-resolver crate instead -- but happy to add a separate crate to your stable if you prefer.

Reply:
I think a separate crate will lend itself to potentially allowing others to reuse the dkim and spf crates independently from the rest of the stuff in this repo. That's not a hard requirement from my perspective, but a nice to have. For the sake of iteration speed, it's fine to keep it gated under a feature in the existing crate, because it can always be extracted later if we need to.

Reply:
Might even be enough to have the `unbound` feature? At least the trait still depends on hickory_proto types so that will need to be a dependency even for the separate trait (but I guess some of the other things wouldn't need to be).

Reply:
Fair -- either way, I think this is now in a good state if you want to merge it.